### PR TITLE
Fix iOS Image Download Bug

### PR DESCRIPTION
**Pull request by 19kmunz**
TLDR: This PR switches from downloadjs to file-saver to enabled shared images for ios support.

[downloadjs](https://www.npmjs.com/package/downloadjs) does not support ios 
>_Devices without file systems like iPhone, iPad, Wii, et al. have nowhere to save the file to, sorry._

Luckily, file-saver does and it is [more widely used](https://npmtrends.com/download-vs-downloadjs-vs-file-saver-vs-file-saverjs-vs-js-file-download). File-saver still complains about how [WebKit sucks (facts) and how they had to do a work around](https://www.npmjs.com/package/file-saver#ios).

I tested [this on a repl.it](https://replit.com/@itsdeviotime/colordle) instance. I would've gone with glitch.com instead but it has a 128MB disk limit, and this app is 500MB of node components. The thief-prevention screen should still remain (based btw), ~feel free to fork it on replit, or clone it locally and Contend With Npm to see the change working.~ Just noticed the vercel integration! Awesome! Ill be sending it to my ios friends to use for now!

Three of my friends and I have been playing colordle since January and sending the pictures + descriptions over discord. Thanks so much for building this!!! i

### CODE REVIEW

File: package-lock.json
Comment (19kmunz) [inline package-lock.json:1]: I am usually a Java developer. The 17k modified lines in package-lock.json TERRIFY me. I just did what i had to do to get this running, (npm pain, node pain, npm audit fix pain, remaking the instance cause audit fix makes things worse pain, pain pain) and This is what changed. I can revert the changes to package-lock, but i am leaving it in cause it isn't in the .gitignore, PLEASE let me know what you think! And PLEASE!! run it on your server on some sort of test url first to double check, if colordle goes down, ill cry!

Comment (hobgoblina) [reply package-lock.json:1]: no worries! 'package-lock.json` helps make sure that builds are reproducible (it's easy for js to result in dependency hell 😛) ... so there could've just been something about your host's environment/OS/architecture/etc that resulted in a build that diverged a decent bit.

probably would've still been fine to deploy, but I tried reverting to the previous package-lock and running `npm i` from my machine, and it resulted in only changing the few things relevant to the switch to the file-saver package. so I went ahead and committed that. 😄

**Review by hobgoblina — APPROVED** [review]
just noticed that this PR existed. glad I caught it within a couple weeks at least.

thanks so much for both catching this issue and getting a fix up!! I wasn't aware that iOS/etc wasn't supported by that package. I sincerely appreciate the effort you put in to figure out an unfamiliar language/environment + test + write a thorough description.

also thanks for the kind words!! I love hearing that people enjoy the game, it really makes my day! 💜

**Comment by 19kmunz** [comment]
@hobgoblina Thanm you for merging!!! It feels good to make a change in something I enjoy, you have made My day 😁 
I wish you a wonderful day, week, month, and year!!

PS: Here is my first saved from palette from the live colordle site 🎉❤️
Guessed colors:
1: Evening Primrose
2: Flamingo Fury
3: Altdorf Guard Blue
4: Caribbean Blue
5: Zima Blue
image
